### PR TITLE
Purifiers get missing features, added ability to make class levels count towards other class levels for specific features only

### PR DESCRIPTION
**Pull request by pheonix99**
TT Purifier technically doesn't lose those revelations, just has forced picks. The level 3 pick is unimplemented and probably unimplementable, so they shouldn't lose it.

They also get armor training as a fighter of four levels less from Celestial Armor, which should include advanced armor training.

Added both and component to enable the second one.

Haven't made armor training -> advanced armor training substitution for them yet.

### CODE REVIEW

**Comment by pheonix99** [comment]
Added crossclass scaling and support for Hellknight and Armored Battlemage armor training.

**Review by Vek17 — CHANGES_REQUESTED** [review]
Can you pull the latest master and rebase this, as well as run it through the linter based on the .editiorconfig on the project. Lots of noise so it is hard to see the actual changes.

File: TabletopTweaks/Bugfixes/Classes/Magus.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+            
+
+            static void PatchArmoredBattlemage()
+            {
+
+                if (ModSettings.Fixes.Fighter.Base.IsDisabled("AdvancedArmorTraining")) { return; }
+                var ArmoredBattlemageArchetype = Resources.GetBlueprint<BlueprintArchetype>("67ec8dcae6fb3d3439e5ae874ddc7b9b");
+
+                Helpers.CreateBlueprint<BlueprintFeature>("ArmoredBattlemageArmorTrainingProgression", x =>
```
[the remark this answers is not in the document]
Comment (pheonix99) [reply TabletopTweaks/Bugfixes/Classes/Magus.cs:46]: Sorry about that on the magus, will move the create blueprint methods over to Added Content as a whole.

Can't figure out how to rebase without making everything go super wierd - i have merged the latest master in.

Could you elaborate on the Linter? I think most of the noise is using Ctrl-K -> Ctrl-D to enforce the default formatting.

[the remark this answers is not in the document]
Comment (pheonix99) [reply TabletopTweaks/Bugfixes/Classes/Magus.cs]: Magus changes made, moved all the bp creates into the armor training file.

[the remark this answers is not in the document]
Comment (pheonix99) [reply TabletopTweaks/Bugfixes/Classes/Magus.cs]: Everything working now playtest is running fine.

**Comment by pheonix99** [comment]
@Vek17 Would it be easier for you to evaluate if I popped open a new branch from master and put the changes in there cleanly, without the mess I made trying to rebase and otherwise make things work?

**Pull request closed without merging**